python/cudf/cudf/core/frame.py

```diff
@@ -550,6 +560,51 @@ def to_cupy(
         -------
         cupy.ndarray
         """
+        if (
+            self._num_columns > 1
+            and na_value is None
+            and self._columns[0].dtype.kind in {"i", "u", "f", "b"}
+            and all(
+                not col.nullable and col.dtype == self._columns[0].dtype
+                for col in self._columns
+            )
+        ):
+            if dtype is None:
+                dtype = np.dtype(self._columns[0].dtype)
+
+            shape = (len(self), self._num_columns)
+            out = cupy.empty(shape, dtype=dtype, order="F")
+
+            table = self.to_pylibcudf()[0]
+            if isinstance(table, plc.Column):
+                table = plc.Table([table])
+            plc.reshape.table_to_array(
+                table,
+                out.data.ptr,
+                out.nbytes,
+            )
+            return out
+        elif self._num_columns == 1:
```

Comment:
Sign post: Nearly identical logic as `SingleColumnFrame`'s `to_cupy` method. But they return cupy arrays with different shapes, so I'm not sure if we can reuse logic.

Reply:
Can you explain further? I am not sure if I understand what is preventing us from using a unified path here. Is it a difference in 1D vs 2D array outputs? Even still, that would just take some wrapping/unwrapping of views and no additional data copies.

Reply:
You're right we can do zero-copy reshape. Done in https://github.com/rapidsai/cudf/pull/18801/commits/366005d559135f06fc88a3a67f2eb995950bc19f